python/do.py

```diff
@@ -0,0 +1,199 @@
+import datetime
+from typing import Any, Dict, Iterator
+
+import mgp
+import gqlalchemy
+from gqlalchemy.connection import _convert_memgraph_value
+
+WRONG_STRUCTURE_MSG = "The `do_case` argument named `conditionals` must be structured as follows: [BOOLEAN, STRING, BOOLEAN, STRING, …​]."
+
+
+@mgp.read_proc
```

Comment:
I think we have a problem with @mgp.read_proc because that means we don't support write queries. Did you maybe intentionally left it?

Reply:
```cypher
MATCH (node:Node {id:12345})
FOREACH (i in CASE WHEN node.needsUpdate THEN [1] ELSE [] END |
  SET node.newProperty = 5678
  REMOVE node.needsUpdate
  SET node:Updated)
...
```
something like this

Reply:
The queries passed to the `case`/`when` functions can be either read or write queries because gqlalchemy opens a new connection independent of the procedure (the `memgraph = gqlalchemy.Memgraph()` line).